src/main/java/com/aws/greengrass/shadowmanager/ShadowManager.java

```diff
@@ -262,6 +259,20 @@ protected void install(InstallConfig installConfig) {
         });
     }
 
+    private void onConfigurationUpdate() {
+        try {
+            componentConfiguration = ComponentConfiguration.from(componentConfiguration, getConfig());
+            configureRateLimits(componentConfiguration.getRateLimitsConfiguration());
+        } catch (InvalidConfigurationException e) {
+            serviceErrored(e);
+        }
+    }
+
+    private void configureRateLimits(RateLimitsConfiguration rateLimitsConfiguration) {
+        inboundRateLimiter.updateRateLimits(rateLimitsConfiguration);
```

Comment:
Don't pass the entire config to the method just pass the values it needs (less coupling)